pkg/util/azure/azure.go

```diff
@@ -34,6 +35,28 @@ func GetHostAlias() (string, error) {
 	return string(all), nil
 }
 
+// GetClusterName returns the name of the cluster containing the current VM
+func GetClusterName() (string, error) {
+	res, err := getResponse(metadataURL + "/metadata/instance/compute/resourceGroupName?api-version=2017-08-01&format=text")
+	if err != nil {
+		return "", fmt.Errorf("Azure ClusterName: unable to query metadata endpoint: %s", err)
```

Comment:
nitpick: No need to prefix your error with `Azure ClusterName:`, that should be handled by the function doing the actual logging. Same for the other functions